Task2

Madiy opened the pull request.

Commit:
added connection to server

Commit:
Connector improvements

Commit:
Connector improvements

Commit:
added request method to Connector

Madiy committed:
Ability of connection to server by name and showing map

Madiy committed:
Ability of connection to server by name and showing map

Commit:
request errors handling & reconnect fix

Commit:
request errors handling & reconnect fix

Commit:
RequestHandler in implementation

Commit:
Connector takes address as tuple

Commit:
Merge branch 'task2' of https://github.com/coder-net/Purple into HEAD

Madiy committed:
Added objects colors,legend and graphDrawer moved to separate file

Special objects on the map are painted in appropriate colors.
Added retractable legend , that shows which color matches what.
Class graphDrawer became big enough and therefore was moved to a separate file.
Some functions was renamed in utils.py

coder-net commented on src/task2/graphDrawer.py:
```diff
@@ -0,0 +1,169 @@
+from PyQt5.QtGui import QPainter, QBrush, QPen
+from PyQt5.QtCore import Qt, QPoint
+
+buildings_color_by_type=[['Town',Qt.red],['Market',Qt.green],['Warehouse',Qt.darkBlue]]
```
here better use `dict`

Madiy replied:
was fixed

coder-net commented on src/task2/server_interface.py:
```diff
@@ -0,0 +1,32 @@
+            return msg
+    
+    def getMapLevel(self,level):
+        if level == 0 or level == 1 or level == 10:
```
here better do `if level in [0, 1, 10]` or `if level in set([0, 1, 10])`

Madiy replied:
was fixed

Madiy committed:
Fixed post coloring &trying in pep8

Previously, nodes were painted by idx number , but not by point_idx - this was mistaken

Madiy committed:
Conversation improvements

Madiy committed:
Getting full path to server_config.json. Some pep8 and server_interface fixes.

When reconnecting or connecting to another user, the old connection is closed "manually".
The connector.request method contained an error filling in the data field - fixed.

Madiy committed:
RequestHandler was not used therefore it was removed

Madiy committed:
Update README.md

coder-net commented on src/task2/connection.py:
```diff
@@ -0,0 +1,129 @@
+    def __init__(self, address: "(SERVER_ADDRESS, SERVER_PORT)" = None):
+        directory = os.path.dirname(os.path.abspath(__file__))
+        if address is None:
+            with open(directory + r"\server_config.json") as json_data:
```
it's not crossplatform
better use `os.path.join` to solve this problem

Madiy replied:
was fixed

Madiy committed:
server_config.json path evaluation fix

coder-net commented on src/task2/connection.py:
```diff
@@ -45,7 +45,7 @@ class Connector:
         directory = os.path.dirname(os.path.abspath(__file__))
         if address is None:
-            with open(directory + r"\server_config.json") as json_data:
+            with open(os.path.join(directory, "server_config.json")) as json_data:
```
чего так сложно кстати?
не знаю как на винде, затести, на линуксе точно можно сделать это вот так
```python
os.path.join('.', 'server_config.json')
```

Madiy replied:
os.path.join('.', 'server_config.json')
Вернёт туже проблему, ради решения которой и делается эта "сложность", а именно:
если указывать относительный ,а не абсолютный путь и производить запуск приложения следующей командой :
'some_place_1'/python 'some_place2'/app.py
server_config.json будет искать в 'some_place1', а не в 'some_place2'

coder-net commented on src/task2/graphDrawer.py:
```diff
@@ -0,0 +1,176 @@
+        for building in self.buildings['posts']:
+            if building['point_idx'] == idx:
+                name = buildings_name_by_number_type[building['type'] - 1]
+                color = buildings_color_by_type[name]
```
некрасиво
1) потому что тебе для каждой вершины нужно проходить по этому списку для поиска текущей вершина, а если у тебя их много, то это может очень сильно замедлять программу
2) мне не совсем нравится твой поиск для цвета здания, тут гораздо проще было бы сделать вот так:
```python
building_type_idx_to_info = {
  1: ("Town", Qt.Color.Red),
  2: ("...", Blue),
  ......,
}
```
всё-таки предлагаю лучше сделать объект `Town`, в котором можно всё хранить и вычислять только при подаче элементов
```python
class Town(QWidget):
    pass
```

coder-net commented on src/task2/utils.py:
```diff
@@ -0,0 +1,124 @@
+
+def graph_to_json(obj, filename):
+    with open(filename) as f:
+        json.dump(obj, filename, default=parse_to_dict)
```
@madiy, ты переименовал метод `parse_to_dict` не везде

coder-net replied:
done

coder-net committed:
fix codestyle and some bugs

coder-net commented on src/task2/app.py:
```diff
@@ -0,0 +1,160 @@
+            self.graphWidget.setWeightLabelsVisible(False)
+            self.graphWidget.setGraph(graph)
+            self.graphWidget.setBuildings(buildings)
+        except BaseException:
```
Так делать не стоит, тем более от `BaseException`, оно же любые исключения перехватывает

coder-net committed:
change point. add Town widget. add Tip to point

Madiy committed:
Error type fix. Unnecessary equating to None removed in server_interface

coder-net committed:
add models for convenience

coder-net committed:
split graph_drawer. delete useless => change working with type in app.py

coder-net committed:
add govnocode. change Building widget and add ability to change size, when window change its size

coder-net committed:
delete useless

coder-net committed:
change scaling of font size for weights

coder-net committed:
Merge pull request #3 from coder-net/change_drawer

Change visualization of vertices
Add models

coder-net commented on src/task2/graph_drawer.py:
```diff
@@ -0,0 +1,164 @@
+
+            painter.end()
+
+    def wheelEvent(self, event):
+        self.zoom += event.angleDelta().y() / 2880
+        self.update()
```
limit `self.zoom`

coder-net commented on src/task2/connection.py:
```diff
@@ -124,106 +124,6 @@ class RequestError(GameConnectionError):
-
 if __name__ == "__main__":
     connector_demonstration()
     send_some_requests()
```
@Madiy, у тебя написано, что ты удаляешь `RequestHandler` в названии коммита
А тут ты удаляешь ещё и функции, которые нужны для проверки
Это ладно, но почему ты тогда не удаляешь вызовы этих функций?
Почему я запускаю этот файл и у меня exception'ы вылетают

coder-net commented on src/task2/server_interface.py:
```diff
@@ -12,14 +12,14 @@ def __init__(self, name):
         self.opened_connection.connect()
         self.opened_connection.send(
             Action.LOGIN, to_json({"name": name}))
-        msg = self.opened_connection.receive()
```
@Madiy, НЕ РАБОТАЕТ!

coder-net committed:
Return used function in connection.py
Uncomment receive data from socket into __init__ of server_interface
